pkg/kcp/provider/gcp/client/provider.go

```diff
@@ -115,5 +119,41 @@ func newHttpClient(ctx context.Context, saJsonKeyPath string) (*http.Client, err
 	if err != nil {
 		return nil, fmt.Errorf("error obtaining GCP HTTP client: [%w]", err)
 	}
+	CheckGcpAuthentication(ctx, client, saJsonKeyPath)
 	return client, nil
 }
+
+func CheckGcpAuthentication(ctx context.Context, client *http.Client, saJsonKeyPath string) {
+	logger := composed.LoggerFromCtx(ctx)
+	cred, err := transport.Creds(ctx, option.WithCredentialsFile(saJsonKeyPath))
+	if err != nil {
+		IncrementCallCounter("Authentication", "Check", "", err)
+		logger.Error(err, "GCP Authentication Check - error obtaining GCP credentials")
+		return
+	}
+	crmService, err := cloudresourcemanager.NewService(ctx, option.WithHTTPClient(client))
+	if err != nil {
+		IncrementCallCounter("Authentication", "Check", "", err)
```

Comment:
Why are we calling IncrementCallCounter only if err != nil? What is the difference between this and other cases where we call IncrementCallCounter regardless of err value?

Reply:
Amir, these errors are from intermediate steps. i.e. they could come when the parsing JSON file failed (or) some issue with creating the GCP CRM Service. I thought we don't need to track the success of intermediate steps as they are part of the checking authentication process. If you look at the line 141, IncrementCallCounter is called even if err != nil.